ir/memory.cpp

```diff
@@ -149,6 +151,20 @@ void Pointer::is_dereferenceable(unsigned bytes, unsigned align) {
   is_dereferenceable(expr::mkUInt(bytes, m.bits_for_offset), align);
 }
 
+// general disjoint check for unsigned integer
```

Comment:
Please add a comment saying that this function assumes that begin1+len1 and begin2+len2 don't overflow (unsigned)